source/grid/manifold_lib.cc

```diff
@@ -419,10 +419,17 @@ SphericalManifold<dim, spacedim>::normal_vector(
   const typename Triangulation<dim, spacedim>::face_iterator &face,
   const Point<spacedim> &                                     p) const
 {
-  // if the maximum deviation for the distance from the vertices to the center
-  // is less than 1.e-5 of the minimum distance to the first vertex, assume we
-  // can simply return p-center. otherwise, we compute the normal using
-  // get_normal_vector
+  // Let us first test whether we are on a "horizontal" or "vertical"
+  // face. In the former case, the normal vector is easy to compute
+  // since it is proportional to the vector from the center to the
+  // point 'p'.
+  //
+  // We test whether that is the case by checking that the vertices
+  // all have roughly the same distance from the center: If the
+  // maximum deviation for the distances from the vertices to the
+  // center is less than 1.e-10 of the distance between vertices (as
```

Comment:
Oh also it is 1.e-5. The 1e-10 is comparing the square of the distances.

Reply:
Ah yes, good catch -- I took the opportunity to also rename the variables to reflect the fact that they represent *squared* distances.